### DISCUSSION TITLE
Persistir lista de Enums

### DISCUSSION THREAD

**Discussion by ivanscardaoni1** [dds-jv/foro#330] [Consultas de persistencia] [answered]
Hola! Estaba haciendo el parcial de Rekomendashi y me surgió una duda.

Tanto en la relación Pedido -> Categoria como en Pedido -> TipoDeCoccion, se presenta una lista de enums en el pedido, y no se me ocurre bien como persistirlas. 

Entiendo que no existe una tabla de enums, sino que cuando es individual, se le pone una anotación de @Enumerated, pero cuando es una lista? Es un @ElementCollection? Si fuese así, el enum ya seria un value object de por si o hay que hacerle alguna anotación? 

Muchas gracias,

Saludos.

**Comment by Franpanozzo** [comment, marked as answer]
Buenas @ivanscardaoni1 , googleando y probando con unos compañeros el mapeo podría ser por ejemplo para las categoriasPreferidas de rekomendashi:

```java
    @EllementCollection
    @OrderColumn(name = "orden")    
    @Enumerated(EnumType.STRING)
    List<Categoria> categoriasPreferidas
```

**Comment by ivanscardaoni1** [comment]
Buenísimo! Ya que esta abierto este hilo, aprovecho para consultar otra cosa del mismo parcial:

En la parte 2 de la sección de persistencia se pide que se pueda pasar un reporte de las 10 piezas mas pedidas cada día de la semana. 
Para eso se me ocurrió persistir el Set en el Pedido (consecuentemente habría que persistir las selecciones), y agregar una fecha al pedido, con eso ya podríamos conseguir las selecciones y con eso la cantidad de piezas por día de semana con un simple algoritmo.

Primero que nada, mi duda principal es si el Set podría ser un value object, ya que no me interesa manejar altas y bajas de sets, simplemente quiero asignarle el set al pedido, porque puede haber mil combinaciones de SETS. Por otro lado quería saber si las selecciones podían ser también value objects por la misma razón. 

Por ultimo queria saber si es necesario especificar el algoritmo que usaría una vez obtengo esos datos, ya que dice "especificar los cambios en el modelo de objetos"
